presto-main/src/main/java/com/facebook/presto/operator/exchange/PartitioningExchanger.java

```diff
@@ -98,6 +87,18 @@ public synchronized void accept(Page page)
         }
     }
 
+    private Page getPartitioningChannelsPages(Page inputPage)
```

Comment:
Not quite understand this method

Reply:
Let's say the input layout is `[data_column_1, data_column_2, partition_column_2, data_column_3, partition_column1]`.
The connector bucket function expects the input layout to only include the `partition_column_N` in the expected order: `[partition_column1, partition_column_2]`
This method extracts partition columns from the input pages in the order defined by the partitioning function.
This method wasn't needed before, as the system partitioning function allows arbitrary pages as it accepts the partitioning channels mask.

Reply:
What about `extractPartitioningColumns` or `extractPartitioningChannels` then?